src/core_plugins/timelion/public/vis/index.js

```diff
@@ -1,35 +1,51 @@
-import { VisVisTypeProvider } from 'ui/vis/vis_type';
+import { VisFactoryProvider } from 'ui/vis/vis_factory';
+import { CATEGORY } from 'ui/vis/vis_category';
 import image from '../images/icon-timelion.svg';
 import { VisTypesRegistryProvider } from 'ui/registry/vis_types';
-import { TemplateVisTypeProvider } from 'ui/template_vis_type';
-import template from 'plugins/timelion/vis/timelion_vis.html';
-import editorTemplate from 'plugins/timelion/vis/timelion_vis_params.html';
-import 'plugins/timelion/vis/timelion_vis_controller';
-import 'plugins/timelion/vis/timelion_vis_params_controller';
-import 'plugins/timelion/vis/timelion_vis.less';
+import { VisRequestHandlersRegistryProvider } from 'ui/registry/vis_request_handlers';
+import { TimelionRequestHandlerProvider } from './timelion_request_handler';
 
-// export the provider so that the visType can be required with Private()
-export default function TimelionVisProvider(Private) {
-  const VisType = Private(VisVisTypeProvider);
-  const TemplateVisType = Private(TemplateVisTypeProvider);
+define(function (require) {
+  // we also need to load the controller and directive used by the template
+  require('plugins/timelion/vis/timelion_vis_controller');
+  require('plugins/timelion/directives/timelion_expression_input');
 
-  // return the visType object, which kibana will use to display and configure new
-  // Vis object of this type.
-  return new TemplateVisType({
-    name: 'timelion',
-    title: 'Timelion',
-    image,
-    description: 'Build time-series using functional expressions',
-    category: VisType.CATEGORY.TIME,
-    template,
-    params: {
-      editor: editorTemplate,
-    },
-    requiresSearch: false,
-    requiresTimePicker: true,
-    implementsRenderComplete: true,
-  });
-}
+  // Stylin
+  require('plugins/timelion/vis/timelion_vis.less');
 
-// register the provider with the visTypes registry so that other know it exists
-VisTypesRegistryProvider.register(TimelionVisProvider);
+  // register the provider with the visTypes registry so that other know it exists
+  VisTypesRegistryProvider.register(TimelionVisProvider);
+  VisRequestHandlersRegistryProvider.register(TimelionRequestHandlerProvider);
```

Comment:
don't agree completely, i think there are reuse scenarios for timelion, but for the time being i can not register the handler (will do so the first time i'll reuse it somewhere)